ivy_models/googlenet/googlenet.py

```diff
@@ -0,0 +1,115 @@
+        v (ivy.Container): Unused parameter. Can be ignored.
+
+    """
+
+    def __init__(
+        self,
+        num_classes=1000,
+        v: ivy.Container = None,
+    ):
+        if v is not None:
+            self.v = v
+        self.num_classes = num_classes
+        super(GoogLeNet, self).__init__(v=v)
+
+    def _build(self, *args, **kwargs):
+        self.conv1 = ConvBlock(3, 64, [7, 7], 2, padding="SAME")
+        self.pool1 = ivy.MaxPool2D([3, 3], 2, "SAME")
+
+        self.conv2 = ConvBlock(64, 64, [1, 1], 1, padding="SAME")
+        self.conv3 = ConvBlock(64, 192, [3, 3], 1, padding="SAME")
+        self.pool3 = ivy.MaxPool2D([3, 3], 2, "SAME")
+
+        self.inception3A = Inception(192, 64, 96, 128, 16, 32, 32)
+        self.inception3B = Inception(256, 128, 128, 192, 32, 96, 64)
+        self.pool4 = ivy.MaxPool2D([3, 3], 2, "SAME")
+
+        self.inception4A = Inception(480, 192, 96, 208, 16, 48, 64)
+
+        self.aux4A = Auxiliary(512, self.num_classes)
+
+        self.inception4B = Inception(512, 160, 112, 224, 24, 64, 64)
+        self.inception4C = Inception(512, 128, 128, 256, 24, 64, 64)
+        self.inception4D = Inception(512, 112, 144, 288, 32, 64, 64)
+
+        self.aux4D = Auxiliary(528, self.num_classes)
+
+        self.inception4E = Inception(528, 256, 160, 320, 32, 128, 128)
+        self.pool5 = ivy.MaxPool2D([3, 3], 2, "SAME")
+
+        self.inception5A = Inception(832, 256, 160, 320, 32, 128, 128)
+        self.inception5B = Inception(832, 384, 192, 384, 48, 128, 128)
+        self.pool6 = ivy.AdaptiveAvgPool2d([1, 1])
+
+        self.dropout = ivy.Dropout(0.4)
+        self.fc = ivy.Linear(1024, self.num_classes)
+
+    def _forward(self, x):
+        out = self.conv1(x)
+        out = self.pool1(out)
+        out = self.conv2(out)
+        out = self.conv3(out)
+        out = self.pool3(out)
+        out = self.inception3A(out)
+        out = self.inception3B(out)
+        out = self.pool4(out)
+        out = self.inception4A(out)
+
+        aux1 = self.aux4A(out)
+
+        out = self.inception4B(out)
+        out = self.inception4C(out)
+        out = self.inception4D(out)
+
+        aux2 = self.aux4D(out)
+
+        out = self.inception4E(out)
+        out = self.pool5(out)
+        out = self.inception5A(out)
+        out = self.inception5B(out)
+
+        out = ivy.permute_dims(out, (0, 3, 1, 2))
+        out = self.pool6(out)
+
+        out = ivy.flatten(out, start_dim=1)
+        out = self.dropout(out)
+        out = self.fc(out)
+
+        return out, aux1, aux2
+
+
+def _inceptionNet_torch_weights_mapping(old_key, new_key):
+    W_KEY = ["conv/weight"]
+    new_mapping = new_key
+    if any([kc in old_key for kc in W_KEY]):
+        new_mapping = {"key_chain": new_key, "pattern": "b c h w -> h w c b"}
+    return new_mapping
+
+
+def inceptionNet_v1(pretrained=True):
+    """InceptionNet-V1 model"""
+    if not pretrained:
```

Comment:
just a small refactor here, as we would want to avoid double model initialisation for better mem usage and efficiency, please ref resnet model for the weights loading code